util/session/sessionmanager.go

```diff
@@ -182,38 +181,13 @@ type standardClaims struct {
 	Subject   string           `json:"sub,omitempty"`
 }
 
-func unixTimeOrZero(t *jwt.Time) int64 {
-	if t == nil {
-		return 0
-	}
-	return t.Unix()
-}
-
 func (mgr *SessionManager) signClaims(claims jwt.Claims) (string, error) {
-	// log.Infof("Issuing claims: %v", claims)
 	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
 	settings, err := mgr.settingsMgr.GetSettings()
 	if err != nil {
 		return "", err
 	}
-	// workaround for https://github.com/argoproj/argo-cd/issues/5217
```

Comment:
@alexmt Since you originally added these in https://github.com/argoproj/argo-cd/pull/5228. Could you help review this removal? Looks like since we are using the new method `jwt.NewNumericDate`, the precision will be in seconds. 

```
// TimePrecision sets the precision of times and dates within this library.
// This has an influence on the precision of times when comparing expiry or
// other related time fields. Furthermore, it is also the precision of times
// when serializing.
//
// For backwards compatibility the default precision is set to seconds, so that
// no fractional timestamps are generated.
var TimePrecision = time.Second

// NumericDate represents a JSON numeric date value, as referenced at
// https://datatracker.ietf.org/doc/html/rfc7519#section-2.
type NumericDate struct {
	time.Time
}

// NewNumericDate constructs a new *NumericDate from a standard library time.Time struct.
// It will truncate the timestamp according to the precision specified in TimePrecision.
func NewNumericDate(t time.Time) *NumericDate {
	return &NumericDate{t.Truncate(TimePrecision)}
}
```

Reply:
I think the workaround that Alex has implemented was due to the fact that our JWTs contain integer timestamps, while the library worked with floats. Now they seem to support multiple precisions and have reverted to integers for the precision we require (seconds), which is nice :) To me, the removal of this workaround looks good.